components/esp_http_server/include/esp_http_server.h

```diff
@@ -1452,6 +1458,66 @@ esp_err_t httpd_queue_work(httpd_handle_t handle, httpd_work_fn_t work, void *ar
  * @}
  */
 
+/* ************** Group: WebSocket ************** */
+/** @name WebSocket
+ * Functions and structs for WebSocket server
+ * @{
+ */
+
+/**
+ * @brief Enum for WebSocket packet types (Opcode in the header)
+ * @note Please refer to RFC6455 Section 5.4 for more details
+ */
+typedef enum {
+    HTTPD_WS_TYPE_CONTINUE   = 0x0,
+    HTTPD_WS_TYPE_TEXT       = 0x1,
+    HTTPD_WS_TYPE_BINARY     = 0x2,
+    HTTPD_WS_TYPE_CLOSE      = 0x8,
+    HTTPD_WS_TYPE_PING       = 0x9,
+    HTTPD_WS_TYPE_PONG       = 0xA
+} httpd_ws_type_t;
```

Comment:
Would it help to have some inbuilt support or helper functions for handling Close, Ping and Pong Control packets?

Reply:
@shahpiyushv I think we can, but leave it as optional in `menuconfig`.

Reply:
@shahpiyushv Same as the async send support as you've mentioned. I think this automatic Ping/Pong support should be implemented but not included in this PR. 

However, if you think we do need this shipped with this PR, I will do it later on next Tuesday after I finish my exam.